mne/io/ant/ant.py

```diff
@@ -157,6 +157,9 @@ def __init__(
             "first_samples": np.array(first_samps),
             "last_samples": np.array(last_samps),
         }
+
+        __import__("inspect").currentframe().f_locals.update(locals())
```

Comment:
Rather than adding this to every module *before* calling `super().__init__(...)`, is there any way this magic could live inside `super.__init__(...)` itself instead? You mention:

> [inspect] only populates the call stack information when an exception occurs

So in theory it should be possible to use `inspect.currentframe().f_back` to get back to this level and do something there, maybe?

Also not sure why you do `__import__("inspect")` rather than just `import inspect` at the top of the file...?